src/App.tsx

```diff
@@ -1,26 +1,121 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useState } from 'react';
+import { getTodos } from './api/todos';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { ErrorMessages } from './types/ErrorTypes';
+import { FilterTypes } from './types/FilterTypes';
+
+import { getVisibleTodos } from './utils/getVisibleTodos';
+import { handleError } from './utils/handleError';
+
+import * as todoService from './api/todos';
+
+import { TodoHeader } from './components/TodoHeader/TodoHeader';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { ErrorPannel } from './components/ErrorPannel/ErrorPannel';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todosToDelete, setTodosToDelete] = useState<number[]>([]);
+  const [selectedFilterType, setSelectedFilterType] = useState<FilterTypes>(
+    FilterTypes.ALL,
+  );
+  const [errorMessage, setErrorMessage] = useState<ErrorMessages>(
+    ErrorMessages.NONE,
+  );
+
+  function handleUpload() {
+    getTodos()
+      .then(setTodos)
+      .catch(() => handleError(setErrorMessage, ErrorMessages.LOAD_FAIL));
+  }
+
+  const isAllCompleted =
+    todos.every(todo => todo.completed) && todos.length !== 0;
+  const filteredTodos = getVisibleTodos(todos, selectedFilterType);
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+  const completedIds = todos
+    .filter(todo => todo.completed)
+    .map(todo => todo.id);
+
+  function handleClearCompleted() {
+    setTodosToDelete(completedIds);
   }
 
+  function deleteTodos(idsToDelete: number[]) {
+    Promise.allSettled(
+      idsToDelete.map(id => {
+        todoService
+          .deleteTodo(id)
+          .then(() => {
+            setTodos(prevTodos => prevTodos.filter(todo => todo.id !== id));
+            setTodosToDelete(() => []);
+          })
+          .catch(() => {
+            setTodos(todos);
+            handleError(setErrorMessage, ErrorMessages.DELETE_FAIL);
+            setTodosToDelete(() => []);
+          });
+      }),
+    );
```

Comment:
Instead of calling setTodos inside each promise's then and catch, gather all results first and update the state once after processing all deletions. This reduces the risk of stale state issues and makes your code cleaner.